cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -109,7 +109,12 @@ protected void verifyData(Id.Application appId,
                             ApplicationSpecification specification) throws DatasetManagementException {
     // NOTE: no special restrictions on dataset module names, etc
     VerifyResult result;
+    MacroChecker macroChecker = new MacroChecker();
     for (DatasetCreationSpec dataSetCreateSpec : specification.getDatasets().values()) {
+      if (macroChecker.isMacro(dataSetCreateSpec.getInstanceName())) {
```

Comment:
yeah, i got confused as even though i was trying to skip creating dataset in plugin when a field is a macro, it tried to create it with macro name and was failing, thinking back i didn't have macro enabled on the field. so its a user error.  ill close this PR.